### PR TITLE
Feature/simulation refactor

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/TimestampChecker.java [hunk 1]
```diff
@@ -62,7 +63,8 @@ public TimestampChecker(Duration acceptableTimeRange) {
 	public Observable<TestInvariantError> check(RunningNetwork network) {
 		return network.ledgerUpdates()
 			.map(Pair::getSecond)
-			.distinct() // Test on only the first ledger update in the network
+			// Test on only the first ledger update in the network
+			.distinct(update -> EpochView.of(update.getTail().getEpoch(), update.getTail().getView()))
 			.filter(l -> !(l.getTail().getEpoch() == 1 && l.getTail().getView().equals(View.of(1))))
```
Comment (siy) [inline radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/TimestampChecker.java:68]: Probably it worth to move filtering condition into dedicated method with descriptive name.

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus/RandomLatencyTest.java [hunk 2]
```diff
@@ -51,10 +52,12 @@
...
-		.checkConsensusLiveness("liveness", synchronousTimeout, TimeUnit.MILLISECONDS)
-		.checkConsensusSafety("safety")
-		.checkConsensusAllProposalsHaveDirectParents("directParents")
-		.checkConsensusNoTimeouts("noTimeouts");
+		.addTestModules(
+			ConsensusMonitors.safety(),
+			ConsensusMonitors.liveness(synchronousTimeout, TimeUnit.MILLISECONDS),
+			ConsensusMonitors.noTimeouts(),
+			ConsensusMonitors.directParents()
+		);
```
Comment (siy) [inline radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus/RandomLatencyTest.java:60]: How about renaming `addTestModules` method and using static imports? Then this part of code might look like
```
.withTest(safety(), liveness(synchronousTimeout, TimeUnit.MILLISECONDS), noTimeouts(), directParents));
```
This is somewhat more concise and more "fluent" for reading, as for me.

**Review by siy — APPROVED** [review]
LGTM with minor comments

**Review by clstrfsck — APPROVED** [review]
Looks good.

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/TimestampChecker.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/TimestampChecker.java:68]: good idea

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus/RandomLatencyTest.java (hunk 2 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus/RandomLatencyTest.java:60]: I think this makes things too abstract for the sake of "fluent" reading. The module the monitor is testing is useful to know as each unit test tests specific modules (e.g. Consensus vs. Ledger).

**Pull request merged**